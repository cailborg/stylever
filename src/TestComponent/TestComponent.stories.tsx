import React from "react";
import { Story, Meta } from "@storybook/react/types-6-0";

import TestComponent from "./TestComponent";

export default {
  title: "Components/TestComponent",
  component: TestComponent,
  argTypes: {
    text: { control: "text" },
  },
} as Meta;

const Template: Story<TestComponentProps> = (args) => (
  <TestComponent {...args} />
);

export const Primary = Template.bind({});
Primary.args = {
  text: "The text of the component",
};
