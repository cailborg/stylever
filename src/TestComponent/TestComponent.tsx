import React from "react";

interface TestComponentProps {
  text: string;
}

const TestComponent: React.FC<TestComponentProps> = ({ text }) => (
  <div>{text}</div>
);

export default TestComponent;
