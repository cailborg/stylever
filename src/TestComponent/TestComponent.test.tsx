import React from "react";
// import { render } from "@testing-library/react";
import { TestComponent } from "./TestComponent";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<TestComponent text="Some text" />).toJSON();
  expect(tree).toMatchSnapshot();
});
