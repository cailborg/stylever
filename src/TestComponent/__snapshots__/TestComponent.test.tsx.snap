// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxjAm ctoFJu"
  data-testid="test-component"
>
  <h1
    className="sc-AxirZ lmNTcb heading"
  >
    I'm the test component
  </h1>
  <h2
    className="sc-AxiKw bOTXee"
  >
    Some text
  </h2>
</div>
`;
